src/redux/services/compare/fetchResources.ts

```diff
@@ -56,7 +70,78 @@ async function fetchResourcesFromCluster(state: RootState): Promise<K8sResource[
     const allYaml = fulfilledResults.map(r => (r as any).value).join(YAML_DOCUMENT_DELIMITER_NEW_LINE);
     return extractK8sResources(allYaml, CLUSTER_DIFF_PREFIX + String(kc.currentContext));
   } catch (err) {
-    log.debug('fetch resources form cluster failed', err);
+    log.debug('fetch resources from cluster failed', err);
+    throw err;
+  }
+}
+
+async function previewHelmResources(state: RootState, options: HelmResourceSet): Promise<K8sResource[]> {
```

Comment:
I wonder how we can re-use this better.. I think the `thunk` and this `listener` should both use one underlying abstraction to fetch resources though it felt a bit out of scope for this. Wdyt?